ElectronClient/gui/KeymapConfigScreen.tsx

```diff
@@ -0,0 +1,229 @@
+import * as React from 'react';
+const CommandService = require('lib/services/CommandService').default;
+const { DOMtoElectronAccelerator } = require('lib/KeymapUtils.js');
+const { themeStyle } = require('lib/theme');
+const { _ } = require('lib/locale.js');
+// const { bridge } = require('electron').remote.require('./bridge');
+// const Setting = require('lib/models/Setting');
+
+interface KeymapItem {
+	label: string,
+	accelerator: string,
+	command: string
+	isEditing: boolean,
+}
+
+// Temporary values
+const commandNames = [
+	'textCopy',
+	'textCut',
+	'textPaste',
+];
+const getAccelerator = (commandName: string) => {
+	switch (commandName) {
+	case 'textCopy':
+		return 'Ctrl+C';
+	case 'textCut':
+		return 'Ctrl+X';
+	case 'textPaste':
+		return 'Ctrl+V';
+	default:
+		throw new Error('Invalid command');
+	}
+};
+
+const initialKeymap = commandNames.map(commandName => {
+	return {
+		label: CommandService.instance().label(commandName),
+		accelerator: getAccelerator(commandName),
+		command: commandName,
+		isEditing: false,
+	};
+});
+
+const KeymapConfigScreen = (props: { theme: Object }) => {
+	const [keymap, setKeymap] = React.useState([...initialKeymap]);
+	const [filter, setFilter] = React.useState('');
+
+	const theme = themeStyle(props.theme);
+	const containerStyle = {
+		...theme.containerStyle,
+		padding: 10,
+		overflow: 'auto',
+	};
+
+	const updateAccelerator = (commandName: string, accelerator: string) => {
+		const _keymap = [...keymap];
+
+		_keymap.find(item => item.command === commandName).accelerator = accelerator;
+		_keymap.find(item => item.command === commandName).isEditing = false;
+		setKeymap(_keymap);
+	};
+
+	const updateIsEditing = (commandName: string, isEditing: boolean) => {
+		const _keymap = [...keymap];
+
+		_keymap.forEach(item => item.isEditing = false);
+		_keymap.find(item => item.command === commandName).isEditing = isEditing;
+		setKeymap(_keymap);
+	};
```

Comment:
You should probably use `useCallback` for this and updateAccelerator otherwise you'll have issues with stale state.

Reply:
Indeed I was putting it at the risk of having a stale state. As a solution, I passed a function to useState(), which changes state by comparing it to previous state. But I don't understand why useCallback() is neccessary to prevent stale state? I'll be glad if you could point me if there's an issue.

Reply:
As I understand, the optimization from useCallback() isn't significant in this case (as there's not many elements here) and will turn out to be an overhead?